Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -363,8 +392,47 @@
         ASSERT_NOT_REACHED();
         return;
     }
+    auto replacedRange = resolvedRange;
+
+    auto subCharacterRange = [](WebCore::CharacterRange superRange, WebCore::CharacterRange previousRange) {
+        auto location = previousRange.location + previousRange.length;
+        return WebCore::CharacterRange(location, superRange.length - previousRange.length);
```

Comment:
what ensures that `superRange.length` is always greater than `previousRange.length?